centaur/src/main/scala/centaur/test/Test.scala

```diff
@@ -772,21 +790,27 @@ object Operations extends StrictLogging {
         Option(CentaurCromwellClient.defaultMetadataArgs.getOrElse(Map.empty) ++ jmArgs))
       jmMetadataObject <- IO.fromTry(Try(jmMetadata.value.parseJson.asJsObject))
       expectation <- IO.fromTry(Try(extractJmStyleMetadataFields(originalMetadata.parseJson.asJsObject)))
-
-      _ <- validateMetadataJson(testType = s"fetchAndValidateJobManagerStyleMetadata (initial fetch)", expectation, jmMetadataObject, submittedWorkflow, workflow)
+      _ <- validateMetadataJson(testType = s"fetchAndValidateJobManagerStyleMetadata", expectation, jmMetadataObject, submittedWorkflow, workflow, allowableOneWordAdditionsInJmMetadata)
     } yield jmMetadata
   }
 
-  val oneWordIncludeKeys = List(
+  val oneWordJmIncludeKeys = List(
     "attempt", "callRoot", "end",
     "executionStatus", "failures", "inputs", "jobId",
     "calls", "outputs", "shardIndex", "start", "stderr", "stdout",
     "description", "executionEvents", "labels", "parentWorkflowId",
     "returnCode", "status", "submission", "subWorkflowId", "workflowName"
   )
 
+  // Our Job Manager metadata validation works by comparing the first pull of metadata which satisfies all test requirements
+  // against what we get when we pull metadata again specifying the job-manager-specific fields.
+  // It's possible for new values to trickle in after the first metadata pull, because our validation wasn't specifically looking for them.
+  // In those cases, the second pull (for Job Manager metadata) may include new values not present in the original.
+  // That's not a sign of an error, so allow additions in these fields when comparing original vs "job-manager only" metadata:
+  val allowableOneWordAdditionsInJmMetadata = List("outputs", "executionEvents", "end")
```

Comment:
Why are these particular keys allowable? I think I know for some but not all; documenting the specifics would probably make this easier to understand.

Reply:
I'll add a little more to the documentation above